packages/thumbprint-react/components/Select/index.tsx

```diff
@@ -23,136 +28,120 @@ const getUIState = ({ isDisabled, hasError }) => {
-    isRequired,
-    name,
-    onChange,
-    onFocus,
-    onBlur,
-    onClick,
-    size,
-    value,
-}) => {
-    const uiState = getUIState({ isDisabled, hasError });
-    return (
-        <div
-            className={classNames({
-                [styles.root]: true,
-                [styles.rootWidthFull]: isFullWidth,
-            })}
-        >
-            <select
-                className={classNames({
-                    [styles.select]: true,
-                    [styles.selectStateDisabled]: uiState === 'disabled',
-                    [styles.selectStateError]: uiState === 'error',
-                    [styles.selectStateDefault]: uiState === 'default',
-                    [styles.selectSizeSmall]: size === 'small',
-                    [styles.selectSizeLarge]: size === 'large',
-                })}
-                id={id}
-                disabled={isDisabled}
-                required={isRequired}
-                value={value}
-                onClick={() => onClick && onClick()}
-                onChange={e => onChange(e.target.value, e)}
-                onFocus={onFocus}
-                onBlur={onBlur}
-                data-test={dataTest}
-                name={name}
-            >
-                {children}
-            </select>
-
-            <NavigationCaretDownSmall className={styles.caret} fill={iconColor[uiState]} />
-        </div>
-    );
-};
-
-Select.propTypes = {
+interface PropTypes {
     /**
      * A collection of [HTML `<option>` elements](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/option).
      */
-    children: PropTypes.node,
+    children?: React.ReactNode;
     /**
      * Adds a HTML `id` attribute to the select. This is used for linking the HTML with a
      * [Label](/components/label/react/).
      */
-    id: PropTypes.string,
+    id?: string;
     /**
      * Visually and functionally disables the select dropdown.
      */
-    isDisabled: PropTypes.bool,
+    isDisabled?: boolean;
     /**
      * Adds `required` HTML attribute to element, indicating that an option with a non-empty string
      * value must be selected.
      */
-    isRequired: PropTypes.bool,
+    isRequired?: boolean;
     /**
      * Makes the radio and text color red.
      */
-    hasError: PropTypes.bool,
+    hasError?: boolean;
     /**
      * Changes the select’s font-size, height, and padding.
      */
-    size: PropTypes.oneOf(['small', 'large']),
+    size?: 'small' | 'large';
     /**
      * Set the `<select>`'s width to `100%` as opposed to the default behavior
      * of only taking up the necessary width.
      */
-    isFullWidth: PropTypes.bool,
+    isFullWidth?: boolean;
     /**
      * The `value` of the `<option>` that is selected. See [React documentation
      * on `<select>` and controlled components](https://facebook.github.io/react/docs/forms.html#the-select-tag)
      * to learn more.
      */
-    value: PropTypes.string.isRequired,
+    value?: string;
```

Comment:
Should this be nullable? The old PropTypes def lists it as required.

Reply:
it should be required, good catch.